lua/entities/gmod_wire_vehicle.lua

```diff
@@ -10,26 +10,34 @@ function ENT:Initialize()
 	self:SetMoveType( MOVETYPE_VPHYSICS )
 	self:SetSolid( SOLID_VPHYSICS )
 
-	self.Inputs = Wire_CreateInputs( self, { "Throttle", "Steering", "Handbrake", "Engine", "Lock" } )
+	self.Inputs = Wire_CreateInputs( self, { "Throttle", "Steering", "Handbrake", "Engine", "Lock", "Vehicle [ENTITY]" } )
+	self.Outputs = Wire_CreateOutputs( self, { "Vehicle [ENTITY]" } )
 end
 
 function ENT:LinkEnt( pod )
 	pod = WireLib.GetClosestRealVehicle(pod,self:GetPos(),self:GetPlayer())
 
 	if not IsValid(pod) or not pod:IsVehicle() then return false, "Must link to a vehicle" end
+	if hook.Run( "CanTool", self:GetPlayer(), WireLib.dummytrace(pod), "wire_vehicle" ) == false then return false, "You do not have permission to access this vehicle" end
+
 	self.Vehicle = pod
 	WireLib.SendMarks(self, {pod})
+	WireLib.TriggerOutput(self, "Vehicle", pod)
 	return true
 end
 function ENT:UnlinkEnt()
 	self.Vehicle = nil
 	WireLib.SendMarks(self, {})
+	WireLib.TriggerOutput(self, "Vehicle", NULL)
 	return true
 end
 
 function ENT:TriggerInput(iname, value)
-	if not IsValid(self.Vehicle) then return end
-	if (iname == "Throttle") then
+	if (iname == "Vehicle") then
+		self:LinkEnt(value)
+	elseif not IsValid(self.Vehicle) then
+		return
```

Comment:
I think the other inputs should still work without a vehicle linked. That way when a vehicle is later linked, they behave as expected.